rust/src/transformations/make_stable_expr/expr_sum/mod.rs

```diff
@@ -0,0 +1,347 @@
+    let margin_id = input_domain.context.grouping_columns();
+    let input_margin = (output_domain.frame_domain.margins.remove(&margin_id))
+        .ok_or_else(|| err!(MakeTransformation, "failed to find margin {:?}", margin_id))?;
+
+    // Set the margins on the output domain to consist of only one margin: 1 row per group, with at most 1 record in each group.
+    let output_margin = input_margin.clone().with_max_partition_size(1);
+    output_domain.frame_domain.margins = HashMap::from_iter([(margin_id, output_margin)]);
+
+    fn stability_map<MI, const P: usize, TI>(
+        series_domain: SeriesDomain,
+        margin: Margin,
+    ) -> Fallible<StabilityMap<MI, LpDistance<P, Scalar>>>
+    where
+        MI: DatasetOuterMetric,
+        MI::Distance: Clone,
+        TI: Summand,
+        Scalar: From<TI::Sum>,
+        f64: InfCast<TI::Sum>,
+    {
+        let (l, u) = series_domain.atom_domain::<TI>()?.get_closed_bounds()?;
+        let (l, u) = (TI::Sum::neg_inf_cast(l)?, TI::Sum::inf_cast(u)?);
+
+        let public_info = margin
+            .public_info
+            .ok_or_else(|| err!(MakeTransformation, "keys must be public information"))?;
+
+        let max_size = usize::exact_int_cast(margin.max_partition_size.ok_or_else(|| {
+            err!(
+                MakeTransformation,
+                "must specify max_partition_len in margin"
+            )
+        })?)?;
+        let pp_relaxation = TI::Sum::relaxation(max_size, l, u)?;
+
+        let pp_map = move |d_in: &IntDistance| match public_info {
+            MarginPub::Keys => TI::Sum::inf_cast(*d_in)?.inf_mul(&l.alerting_abs()?.total_max(u)?),
+            MarginPub::Lengths => TI::Sum::inf_cast(*d_in / 2)?.inf_mul(&u.inf_sub(&l)?),
+        };
+
+        Ok(StabilityMap::new_fallible(move |d_in: &MI::Distance| {
+            let (l0, l1, li) = MI::triple(d_in.clone());
+            let (l0, l1, li) = (TI::Sum::inf_cast(l0)?, pp_map(&l1)?, pp_map(&li)?);
+
+            let mcp = MI::max_changed_partitions(d_in.clone(), margin.max_num_partitions.clone())?;
+            let relaxation = TI::Sum::inf_cast(mcp)?.inf_mul(&pp_relaxation)?;
+
+            Ok(if P == 1 {
+                let ideal_sens = l1.total_min(l0.inf_mul(&li)?)?;
+                Scalar::from(ideal_sens.inf_add(&relaxation)?)
+            } else if P == 2 {
+                let (l0, l1, li) = (f64::inf_cast(l0)?, f64::inf_cast(l1)?, f64::inf_cast(li)?);
+                let relaxation = f64::inf_cast(relaxation)?;
+                let ideal_sens = l1.total_min(l0.inf_sqrt()?.inf_mul(&li)?)?;
+                Scalar::F64(ideal_sens.inf_add(&relaxation)?)
+            } else {
+                return fallible!(MakeTransformation, "unsupported Lp norm");
+            })
+        }))
+    }
+
+    let series_domain = input_domain.active_series()?.clone();
+    let stability_map = match dtype {
+        // DataType::UInt8 => stability_map::<MI, P, u8>(series_domain, margin),
```

Comment:
Is this a TODO? Explain.

Reply:
u8 and u16 are opt-in for Polars, and I haven't opted in. I'll leave them off unless someone asks for them.